cmd/api/src/api/v2/saved_queries.go

```diff
@@ -100,3 +104,26 @@ func (s Resources) CreateSavedQuery(response http.ResponseWriter, request *http.
 		api.WriteBasicResponse(request.Context(), savedQuery, http.StatusCreated, response)
 	}
 }
+
+func (s Resources) DeleteSavedQuery(response http.ResponseWriter, request *http.Request) {
+	var (
+		rawSavedQueryID = mux.Vars(request)[api.URIPathVariableSavedQueryID]
+		userID          = ctx2.FromRequest(request).AuthCtx.Session.UserID
+	)
+
+	if savedQueryID, err := strconv.Atoi(rawSavedQueryID); err != nil {
+		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, api.ErrorResponseDetailsIDMalformed, request), response)
+	} else if savedQueryBelongsToUser, err := s.DB.SavedQueryBelongsToUser(userID, savedQueryID); err != nil {
+		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request), response)
+	} else if !savedQueryBelongsToUser {
+		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, "invalid saved_query_id supplied", request), response)
+	} else if err := s.DB.DeleteSavedQuery(savedQueryID); err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
We do not want gorm leaking into the API layer, so this should be doing a check against `database.ErrNotFound`. It's still not ideal to have database types in our API layer either, but it's less of a problem than leaking `gorm` and is why `CheckError()` is used in the first place